### DISCUSSION TITLE
C++をSageMaker Studio Labで使いたい

### DISCUSSION THREAD

**Discussion by icoxfog417** [aws-sagemaker-jp/awesome-studio-lab-jp#13] [Q&A] [answered]
C++をSageMaker Studio Labで実行する方法はあるでしょうか?

**Comment by icoxfog417** [comment, marked as answer]
あります。JupyterはKernelを追加することで他の言語を動かすことができます。以下はその言語の一覧ですが、C++は拡張可能な言語のひとつになります。

[Jupyter kernels](https://github.com/jupyter/jupyter/wiki/Jupyter-kernels)

C++は[xeus-cling](https://github.com/QuantStack/xeus-cling)でKernelを追加できることを確認しました。

1. ターミナルから下記コマンドで`cpp`の環境を作成
   * `conda install xeus-cling libstdcxx-devel_linux-64==9.4.0 libgcc-devel_linux-64==9.4.0 -c conda-forge`
   * [※2022年10月時点だと、9.4.0を指定しないとC++14/17が動かないようです](https://github.com/jupyter-xeus/xeus-cling/issues/415#issuecomment-1227200351)
2. JupyterLabを再起動
   * ![image](https://user-images.githubusercontent.com/544269/193822726-8f8a6707-7573-4740-b158-0e07ecff84ab.png)
3. C++のカーネルが追加されていることを確認
   * ![image](https://user-images.githubusercontent.com/544269/193822158-8ed0b2e8-147d-4916-bfb3-5ea6e1c04430.png)

行列演算を行う場合は[xtensor](https://github.com/xtensor-stack/xtensor)が便利です。

**Reply by yukiesuta** [reply]
現時点ではconda install xeus-cling libstdcxx-devel_linux-64==9.4.0 libgcc-devel_linux-64==9.4.0 -c conda-forgeでカーネルが追加されません、、対処法はありますか？
